test/desdemona/cli_test.clj

```diff
@@ -88,3 +117,54 @@
         (is (thrown-with-msg?
              Exception (re-pattern @#'aeron/aeron-launch-error-message)
              (with-fake-launcher-side-effects (aeron/-main))))))))
+
+(deftest peers-main-tests
+  (testing "first argument must be an integer"
+    (is (thrown-with-msg?
+         NumberFormatException #"BOGUS"
+         (peers/-main "BOGUS"))))
+  (testing "happy case"
+    (let [n-peers 6
+          group (gensym)
+          peers (for [_ (range n-peers)] (gensym))
+          events (atom [])
+          fn-results {'onyx.api/start-peer-group group
+                      'onyx.api/start-peers peers}
+          redef-pairs (for [sym ['onyx.api/start-peer-group
+                                 'onyx.api/start-env
+                                 'onyx.api/start-peers
+                                 'desdemona.launcher.utils/add-shutdown-hook!
+                                 'onyx.api/shutdown-peer
+                                 'onyx.api/shutdown-peers
+                                 'onyx.api/shutdown-peer-group
+                                 'clojure.core/shutdown-agents]]
+                        [(resolve sym)
+                         (fn [& args]
+                           (let [event (into [sym] args)]
+                             (swap! events conj event)
+                             (fn-results sym)))])
+          redefs (into {} redef-pairs)]
+      (with-redefs-fn redefs
+        (fn []
+          (is (= [] @events))
+          (let [[result stdout] (with-fake-launcher-side-effects
+                                  (peers/-main (str n-peers)))]
+            (let [before-events @events]
+              (is (= [['onyx.api/start-peer-group expected-peer-config]
+                      ['onyx.api/start-env expected-env-config]
+                      ['onyx.api/start-peers n-peers group]]
+                     (butlast before-events)))
```

Comment:
I'd just equality check, but I can't produce something that's equal to that fn, which is why I have to destructure that one separately.